### PR TITLE
feat/CCN3-1825/nannies-update-personal-details-task-for-known-children

### PR DESCRIPTION
**Pull request by proctorj**
## Description

This PR adds a new page to the 'personal details' task where a user is asked if they have any of their own children who are under 16. The Boolean field allows for the conditional reveal of the 'your children' task within the task list. As the contents of this task are outside of the ticket specification, the task is currently a mirror of the 'childcare address' task. The Tasklist has been refactored and a hide/show function created.

## Todo's before PR

- [x] Branch has been rebased against develop
- [x] Unit tests passed (`make test`)
- [ ] Selenium tests have passed
- [x] PR named appropriately - see [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Commits-guideline)
- [x] PR description describes the overall goals of PR commits
- [x] Templates have been checked against the relevant user-story

## Migrations

- [x] Null fields in models specifies default value
- [x] You generated and applied migrations (`make migrate`)
- [ ] You updated fixtures (`make export`)

## PR Todo's

Please refer to PR [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-submit-a-pull-request)

- [x] Specified reviewers (even if it's yourself)
- [x] Specified assignees (those who'll merge)
- [x] Specified labels

## PR review

Please refer to PR review [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-review-a-pull_request)

- [ ] Code syntax formatting checked
- [ ] Debug messages are absent

## PR merge

Select only one:

- [ ] Should be merged?
- [x] Should be rebased?
- [ ] Should be squashed?

Please refer to PR merge [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-merge-a-pull_request)

### CODE REVIEW

**Comment by ricaduchateau** [comment]
rerun jenkins build

File: personal_details_app/tests/test_views/test_your_children.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+from django.urls import resolve
+from unittest import mock
+from ...views import *
+import uuid
+from django.template.response import TemplateResponse
+
+from nanny.test_utils import side_effect
+
+
+@mock.patch("nanny.db_gateways.IdentityGatewayActions.read", authenticate)
```
Comment (LRydeard) [inline personal_details_app/tests/test_views/test_your_children.py]: I've never seen a patch decorator on a class before.
What does this decorator do?

File: personal_details_app/views/your_children.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    def form_valid(self, form):
+
+        application_id = app_id_finder(self.request)
+        application_record = NannyGatewayActions().read('application', params={'application_id': application_id}).record
+        if application_record['personal_details_status'] != 'COMPLETED' or application_record['personal_details_status'] != 'FLAGGED':
+            application_record['personal_details_status'] = 'IN_PROGRESS'
+        NannyGatewayActions().put('application', params=application_record)
+
+        nanny_api_response = NannyGatewayActions().read('application', params={'application_id': application_id})
+        if nanny_api_response.status_code == 200:
```
Comment (LRydeard) [inline personal_details_app/views/your_children.py]: This code seems redundant, could you review it?

File: personal_details_app/views/your_children.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            'your_children': form.cleaned_data['your_children'],
+        }
+
+        existing_record = NannyGatewayActions().read('applicant-personal-details', params={'application_id': application_id})
+        if existing_record.status_code == 200:
+            NannyGatewayActions().patch('applicant-personal-details', params=data_dict)
+        elif existing_record.status_code == 404:
+            NannyGatewayActions().create('applicant-personal-details', params=data_dict)
+
+        self.success_url = 'personal-details:Personal-Details-Summary'
```
Comment (LRydeard) [inline personal_details_app/views/your_children.py]: Why are you setting the success_url to the same value here?

File: tasks_app/templates/task-list.html [hunk 4]
```diff
@@ -20,72 +20,76 @@ <h1 class="form-title heading-large">Register as a nanny</h1>
...
-
-                    {% endif %}
-
-                {% endif %}
-
-                </a>
-            </td>
-        </tr>
-    {% endfor %}
+                        {% if task.name == "review" and all_complete == False %}
```
Comment (LRydeard) [inline tasks_app/templates/task-list.html]: Could you fix the indentation on this code block?

**Comment by ricaduchateau** [comment]
rerun jenkins build

**Comment by proctorj** [comment]
@LRydeard Could you take another look at this when you get some time?

**Pull request merged**